src/widget/wtrackmenu.cpp

```diff
@@ -1595,6 +1625,161 @@ void WTrackMenu::slotClearAllMetadata() {
+        QListWidget* delListWidget = new QListWidget;
+        delListWidget->setSizePolicy(QSizePolicy(QSizePolicy::Minimum,
+                QSizePolicy::MinimumExpanding));
+        delListWidget->addItems(locations);
+        delListWidget->setMinimumSize(
+                mixxx::widgethelper::adjustedListWidgetSize(*delListWidget, *this));
+
+        QDialogButtonBox* delButtons = new QDialogButtonBox();
+        QPushButton* cancelBtn = delButtons->addButton(
+                tr("Cancel"),
+                QDialogButtonBox::RejectRole);
+        QPushButton* deleteBtn = delButtons->addButton(
+                tr("Delete Files"),
+                QDialogButtonBox::AcceptRole);
+        cancelBtn->setDefault(true);
+        // This is required after customizing the buttons, otherwise neither button
+        // would close the dialog.
+        connect(cancelBtn, &QPushButton::clicked, dlgDelConfirm, &QDialog::reject);
+        connect(deleteBtn, &QPushButton::clicked, dlgDelConfirm, &QDialog::accept);
+
+        delLayout->addWidget(delListWidget);
+        delLayout->addWidget(delWarning);
+        delLayout->addWidget(delButtons);
+        dlgDelConfirm->setLayout(delLayout);
+
+        if (dlgDelConfirm->exec() == QDialog::Rejected) {
+            return;
+        }
+    }
+
+    // Set up and initiate the track batch operation
+    const auto progressLabelText =
+            tr("Removing %1 track file(s) from disk...",
+                    "",
+                    getTrackCount());
+    const auto trackOperator =
+            RemoveTrackFilesFromDiskTrackPointerOperation();
+    applyTrackPointerOperation(
+            progressLabelText,
+            &trackOperator);
+
+    // Purge deleted tracks and show deletion summary message.
+    if (trackOperator.tr_tracksToPurge.length() > 0) {
+        // Purge only those tracks whose files were actually deleted.
+        m_pLibrary->trackCollectionManager()->purgeTracks(trackOperator.tr_tracksToPurge);
+        // Optional message box:
+        QMessageBox msgBoxPurgeTracks(QMessageBox::Information,
+                QObject::tr("Track files deleted"),
+                nullptr);
+        msgBoxPurgeTracks.setText(
+                tr("%1 track files were deleted from disk and purged "
+                   "from the Mixxx database.")
+                        .arg(QString::number(
+                                trackOperator.tr_tracksToPurge.length())) +
+                QString("<br><br>") +
+                tr("Note: if you are in Browse or Recording you need to "
+                   "click the current view again to see changes."));
+        msgBoxPurgeTracks.setTextFormat(Qt::RichText);
+        msgBoxPurgeTracks.exec();
+    }
+
+    if (trackOperator.tr_tracksToKeep.length() >= 0) {
+        return;
+    }
+
+    {
+        // If there are tracks that could not be deleted show a message with those listed.
+        QDialog* dlgNotDeleted = new QDialog(nullptr);
```

Comment:
dlgNotDeleted can be allocated on the stack, no need to use `new`. Your current version leaks memory.

Reply:
alright.